databricks/koalas/internal.py

```diff
@@ -698,7 +698,7 @@ def spark_columns(self) -> List[spark.Column]:
         index_spark_columns = self.index_spark_columns
```

Comment:
Shall we add some docstrings clearly to describe when to use which clearly?

For example, `spark_frame` now should always be used via `df._internal.applied.spark_frame` for Spark DataFrame APIs that internally creates new query execution plan.

For expressions and/or functions, `df._internal.spark_frame` should be used together with Spark column instances, in order to avoid the operations on different DataFrames.

Reply:
Maybe we could also document that `spark_frame` is just the base Spark DataFrame where the expressions and functions are not applied.

We might have to consider mentioning about:
- Spark expressions/functions, to create new Spark Columns against the same DataFrame.
- Spark DataFrame APIs that internally creates query execution plans, to create a new DataFrame.